cwms-data-api/src/main/java/cwms/cda/api/timeseriesprofile/TimeSeriesProfileParserCreateController.java

```diff
@@ -0,0 +1,100 @@
+ */
+
+package cwms.cda.api.timeseriesprofile;
+
+import static cwms.cda.api.Controllers.CREATE;
+import static cwms.cda.api.Controllers.FAIL_IF_EXISTS;
+import static cwms.cda.data.dao.JooqDao.getDslContext;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import cwms.cda.data.dao.timeseriesprofile.TimeSeriesProfileParserDao;
+import cwms.cda.data.dto.timeseriesprofile.TimeSeriesProfileParserColumnar;
+import cwms.cda.data.dto.timeseriesprofile.TimeSeriesProfileParserIndexed;
+import cwms.cda.formatters.Formats;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiRequestBody;
+import javax.servlet.http.HttpServletResponse;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.DSLContext;
+
+
+public final class TimeSeriesProfileParserCreateController extends TimeSeriesProfileParserBase implements Handler {
+
+    public TimeSeriesProfileParserCreateController(MetricRegistry metrics) {
+        tspMetrics(metrics);
+    }
+
+    @OpenApi(
+        queryParams = {
+            @OpenApiParam(name = FAIL_IF_EXISTS, type = boolean.class, description = "If true, the parser will"
+                + " fail to save if the TimeSeriesProfileParser already exists"),
+        },
+        requestBody = @OpenApiRequestBody(
+            content = {
+                @OpenApiContent(from = TimeSeriesProfileParserIndexed.class, type = Formats.JSONV1),
+                @OpenApiContent(from = TimeSeriesProfileParserColumnar.class, type = Formats.JSONV1),
+            },
+            required = true
+        ),
+        method = HttpMethod.POST,
+        summary = "Parse TimeSeriesProfileData into a new Time Series Profile",
+        tags = {TAG}
+    )
+    @Override
+    public void handle(@NotNull Context ctx) {
+        try (final Timer.Context ignored = markAndTime(CREATE)) {
+            DSLContext dsl = getDslContext(ctx);
+
+            boolean failIfExists = ctx.queryParamAsClass(FAIL_IF_EXISTS, Boolean.class).getOrDefault(true);
+
+            TimeSeriesProfileParserDao tspParserDao = getParserDao(dsl);
+            String content = ctx.body();
+            if (content.isEmpty()) {
+                ctx.status(HttpServletResponse.SC_BAD_REQUEST);
+                return;
+            }
+            if (content.contains(COLUMNAR_TYPE)) {
+                TimeSeriesProfileParserColumnar tspParserColumnar = Formats
+                        .parseContent(Formats.parseHeader(Formats.JSONV1, TimeSeriesProfileParserColumnar.class),
+                                content, TimeSeriesProfileParserColumnar.class);
+                tspParserDao.storeTimeSeriesProfileParser(tspParserColumnar, failIfExists);
+            } else {
```

Comment:
The error handling on else here can be confusing, if the user provided neither type they will receive an error about the parser indexed type.

Reply:
Changed to an `else if` and added a default case that responds with a bad request error.